### PR TITLE
Compact display of geometry without WKT

### PR DESCRIPTION
**Pull request by evetion**
Getting the WKT of a (complex) geometry is non trivial and will slow down things especially when used in Tables.

With this PR:

When used in a Table (compact)
```julia
julia> df = DataFrame(geom=createpoint.(coords), name="test")
10×2 DataFrame
 Row │ geom                name
     │ IGeometr…           String
─────┼────────────────────────────
   1 │ Geometry: wkbPoint  test
   2 │ Geometry: wkbPoint  test
   3 │ Geometry: wkbPoint  test
   4 │ Geometry: wkbPoint  test
   5 │ Geometry: wkbPoint  test
   6 │ Geometry: wkbPoint  test
   7 │ Geometry: wkbPoint  test
   8 │ Geometry: wkbPoint  test
   9 │ Geometry: wkbPoint  test
  10 │ Geometry: wkbPoint  test
```

Individual or array (non-compact) usage
```julia
julia> df[1, :geom]
Geometry: POINT (0.394776848700807 0.244117475869569)
```

### CODE REVIEW

**Comment by evetion** [comment]
We could also completely disable WKT printing?

**Comment by visr** [comment]
Could you add a test here (with/without compact) and earn that coverage increase? ;)
https://github.com/yeesian/ArchGDAL.jl/blob/f56f79c23b47e43d968605b053a531cf902991ae/test/test_display.jl#L66-L67

Otherwise this looks good to me. Not sure why we'd want to disable WKT printing altogether.

**Comment by evetion** [comment]
> Otherwise this looks good to me. Not sure why we'd want to disable WKT printing altogether.

It's unasked for and it's non-constant in time/memory. It's a "nice" thing that works well for a `Point(1,2)`, but quickly becomes slow for a large, complex geometry.

I would agree it's better than just saying `IGeometry`, but that's why I made #160.

File: test/test_display.jl [hunk 1]
```diff
@@ -19,6 +19,8 @@ import ArchGDAL; const AG = ArchGDAL
              Field 0 (FID): [OFTReal], 2.0, 3.0, 0.0, 3.0
              Field 1 (pointname): [OFTString], point-a, point-b, a, b
         """
+        @test sprint(print, AG.nextnamedtuple(layer), context=:compact => true) == 
+        "(FID = 2.0, pointname = \"point-a\",  = Geometry: wkbPoint)"
```
Comment (visr) [inline test/test_display.jl:23]: Thought this looks funny, but that's an empty geometry name of course.

**Review by visr — APPROVED** [review]

File: test/test_display.jl (hunk 1 above)
Comment (evetion) [reply test/test_display.jl:23]: Yeah, I was suprised a `Symbol("")` was actually allowed as key in the beginning.

**Pull request merged**